test/libyul/objectCompiler/identical_subobjects_partial_debug_info_no_use_src.yul

```diff
@@ -0,0 +1,169 @@
+// All objects have identical unoptimized code, but with different debug annotations.
+// After optimizations we should end up with identical code for all of them, because
+// debug info will be stripped due to missing @use-src annotations.
```

Comment:
How do you justify this behavior?
This seems very problematic - I thought you had a version in which this kind of thing *didn't* happen in the end?

Reply:
I guess the description is just bad in this comment - the point is that the sources here do **not** have valid source location comments, since they will *never* be parsed and used without a ``@use-src`` annotation, right?

Reply:
I.e. there is nothing stripped here, there aren't any valid source locations in the first place, so they don't affect the caching is rather what's happening, I'd say.

Reply:
It's just a matter of point of view. I was thinking about the comments in the source and apparently you're thinking about debug data in the AST instead.

I meant that location comments are not the same in the sources. I.e. the objects don't all contain the `@src` annotation. I guess if you're only looking at what ends up in the parsed AST then yeah, you could say there's no debug info.

Similarly, the original source contains location comments and optimized IR does not. So from that point of view they get stripped.

Anyway, I reworded this to say "location comments" rather than "debug info" when I specifically mean what's in the source. Hope that makes things clearer.